flexmeasures_openweathermap/cli/commands.py

```diff
@@ -1,17 +1,148 @@
 from flask import current_app
+
 from flask.cli import with_appcontext
 import click
+from flexmeasures.data.models.time_series import Sensor
+
 from flexmeasures.data.transactional import task_with_status_report
+from flexmeasures.data.config import db
+
+from .. import flexmeasures_openweathermap_bp
+from .schemas.weather_sensor import WeatherSensorSchema
+from ..utils.modeling import (
+    get_or_create_weather_station,
+)
+from ..utils.locating import get_locations
+from ..utils.filing import make_file_path
+from ..utils.owm import (
+    save_forecasts_in_db,
+    save_forecasts_as_json,
+    get_supported_sensor_spec,
+)
+from ..sensor_specs import owm_to_sensor_map
 
-from .. import flexmeasures_openweathermap_cli_bp
 
+"""
+TODO: allow to also pass an asset ID or name for the weather station (instead of location) to both commands?
+"""
 
-@flexmeasures_openweathermap_cli_bp.cli.command("hello-world")
+supported_sensors_list = ", ".join([str(o["name"]) for o in owm_to_sensor_map.values()])
+
+
+@flexmeasures_openweathermap_bp.cli.command("register-weather-sensor")
+@with_appcontext
 @click.option(
     "--name",
+    required=True,
+    help=f"Name of the sensor. Has to be from the supported list ({supported_sensors_list})",
 )
+# @click.option("--generic-asset-id", required=False, help="The asset id of the weather station (you can also give its location).")
+@click.option(
+    "--latitude",
+    required=True,
+    type=float,
+    help="Latitude of where you want to measure.",
+)
+@click.option(
+    "--longitude",
+    required=True,
+    type=float,
+    help="Longitude of where you want to measure.",
+)
+@click.option(
+    "--timezone",
+    default="UTC",
+    help="The timezone of the sensor data as string, e.g. 'UTC' (default) or 'Europe/Amsterdam'",
+)
+def add_weather_sensor(**args):
+    """
+    Add a weather sensor.
+    This will first create a weather station asset if none exists at the location yet.
+
+    """
+    errors = WeatherSensorSchema().validate(args)
+    if errors:
+        click.echo(
+            f"[FLEXMEASURES-OWM] Please correct the following errors:\n{errors}.\n Use the --help flag to learn more."
+        )
+        raise click.Abort
+
+    weather_station = get_or_create_weather_station(args["latitude"], args["longitude"])
+    args["generic_asset"] = weather_station
+    del args["latitude"]
+    del args["longitude"]
+
+    fm_sensor_specs = get_supported_sensor_spec(args["name"])
+    args["event_resolution"] = fm_sensor_specs["event_resolution"]
+    args["unit"] = fm_sensor_specs["unit"]
+    sensor = Sensor(**args)
```

Comment:
More clear would be to:
- use `Sensor(**fm_sensor_specs)` in line 78
- delete lines 72, 73, 76, 77 and 79 (for line 79, see my comment in `sensor_specs` about renaming the key "seasonality" to "attributes")
- move line 71 to between 75 and 78, and edit it to become `fm_sensor_specs["generic_asset"] = weather_station`
- Copy the timezone info from the weather station to the sensor, just before 78: `fm_sensor_specs["timezone"] = weather_station.timezone`